packages/nextjs/src/index.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Fedify with Next.js
+ * ================
+ *
+ * This module provides a [Next.js] middleware to integrate with the Fedify.
+ *
+ * [Next.js]: https://nextjs.org/
+ *
+ * @module
+ * @since 1.9.0
+ */
+import type { Federation, FederationFetchOptions } from "@fedify/fedify";
+import { notFound } from "next/navigation";
+import { NextResponse } from "next/server";
+import { getXForwardedRequest } from "x-forwarded-fetch";
+
+interface ContextDataFactory<TContextData> {
+  (request: Request):
+    | TContextData
+    | Promise<TContextData>;
+}
+type ErrorHandlers = Omit<FederationFetchOptions<unknown>, "contextData">;
+
+/**
+ * Wrapper function for Next.js middleware to integrate with the
+ * {@link Federation} object.
+ *
+ * @template TContextData A type of the context data for the
+ *                         {@link Federation} object.
+ * @param {Federation<TContextData>} federation A {@link Federation} object
+ *                         to integrate with Next.js.
+ * @param {ContextDataFactory<TContextData>} contextDataFactory A function
+ *                         to create a context data for the
+ *                         {@link Federation} object.
+ * @param {Partial<ErrorHandlers>} errorHandlers A set of error handlers to
+ *                         handle errors during the federation fetch.
```

Comment:
Parameter types can be omitted:

```suggestion
 * @param federation A {@link Federation} object
 *                   to integrate with Next.js.
 * @param contextDataFactory A function to create a context data for the
 *                          {@link Federation} object.
 * @param errorHandlers A set of error handlers to
 *                      handle errors during the federation fetch.
```